### PR TITLE
Ifrost/1945970 cad page scaffolding

### PR DESCRIPTION
**Pull request by ifrost1**
### Summary:
- Scaffolded out CAD page
- Adjusted `Layout` pattern to accommodate wider sidebar
- Added `squared` and `flush` variants of the `Panel` component
- Created dynamic `OverflowList` component that shows and hides shadows to help with accessibility by calculating the width of the items inside. Could be reused with other elements.
- Made page header more dynamic with right slot that is more responsive.

### Test Plan:

`yarn lint` runs without errors
`yarn test` runs without errors
`yarn types` runs without errors.

### CODE REVIEW

File: src/components/PageHeader/PageHeader.tsx [hunk 1]
```diff
@@ -57,14 +63,18 @@ export const PageHeader = ({
     <header className={componentClassName} {...other}>
       <div className={styles['page-header__left']}>
```
Comment (jinlee93) [inline src/components/PageHeader/PageHeader.tsx:64]: Looking at https://www.figma.com/file/0jYcObnngcbsiBOsOPUmtR/Implementation-H1-'22?node-id=2457%3A118577 and https://www.figma.com/file/yLYL19XZI0SWhbQJYuKVso/EDS-Pilot%3ACAD?node-id=1190%3A17197 I think the left and right sections are supposed to take up 50% width each for desktop?

Comment (jinlee93) [reply src/components/PageHeader/PageHeader.tsx:64]: And the button group would be aligned to the right

Comment (ifrost1) [reply src/components/PageHeader/PageHeader.tsx:64]: Let's get all of the content (like your stepper) in there first and then we can adjust if needed. One of those things that is hard to scope out without everything in there. Totally hear ya that we may need a 50/50 approach there. But let's get something in there and then we can adjust if that's alright

Comment (ifrost1) [reply src/components/PageHeader/PageHeader.tsx:64]: @jinlee93  Let's get all of the content (like your stepper) in there first and then we can adjust if needed. One of those things that is hard to scope out without everything in there. Totally hear ya that we may need a 50/50 approach there. But let's get something in there and then we can adjust if that's alright

Comment (jinlee93) [reply src/components/PageHeader/PageHeader.tsx:64]: That sounds good to me

**Pull request merged**